packages/graph/hash_graph/lib/graph/src/types/schema/link.rs

```diff
@@ -0,0 +1,91 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::types::{
+    schema::{
+        array::{Array, MaybeOrdered},
+        combinator::Optional,
+        ValidationError,
+    },
+    Uri,
+};
+
+/// Intermediate representation used during deserialization.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct LinksRepr {
+    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
+    links: HashMap<Uri, Optional<MaybeOrdered<Array>>>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    required_links: Vec<Uri>,
+}
+
+/// Links to entities.
```

Comment:
I'm not sure how you're writing the docs for these schemas but I think this one is very misleading and the "schema" bit shouldn't be implicit. This struct is defining the allowed (and required) Link Types that are defined through an Entity _Type_. These constrain the links that can be present on an entity of that type. 

It is **not** a collection of links to _entities_.